vmm/src/memory_manager.rs

```diff
@@ -733,12 +748,20 @@ impl Aml for MemoryManager {
                     // I/O port for memory controller
                     &aml::Name::new(
                         "_CRS".into(),
-                        &aml::ResourceTemplate::new(vec![&aml::IO::new(
-                            0x0a00, 0x0a00, 0x01, 0x18,
+                        &aml::ResourceTemplate::new(vec![&aml::AddressSpace::new_memory(
+                            aml::AddressSpaceCachable::NotCacheable,
+                            true,
+                            self.device_base.0,
+                            self.device_base.0 + MemoryManager::DEVICE_SIZE - 1,
```

Comment:
This is exposing a bit of an ACPI quirk: when creating the address space you specify the minimum address (it could be movable, but we never do that), the maximum (which is the last addressable byte) and the length. The ACPI compiler will even complain if the min + length - 1 is less than the max. With retrospect I should have designed this constructor to take minimum and length.

Reply:
Ok thanks for clarification. ACPI seems a bit stupid then. Why does it need this field if it knows the start, the size and complain about the last not being start+size-1?

Reply:
It's because it *could* be movable so (max - min) > length. I've never seen it in practice so i've just implemented the constructor with 2 of the 3 fields and calculate the third

Reply:
Ok got it!